src/analysis/special_functions/trigonometric.lean

```diff
@@ -2205,6 +2236,47 @@ begin
     ring },
 end
 
+lemma tan_add {x y : ℂ}
+  (h : ((∀ k : ℤ, x ≠ (2 * k + 1) * π / 2) ∧ ∀ l : ℤ, y ≠ (2 * l + 1) * π / 2)
+     ∨ ((∃ k : ℤ, x = (2 * k + 1) * π / 2) ∧ ∃ l : ℤ, y = (2 * l + 1) * π / 2)) :
+  tan (x + y) = (tan x + tan y) / (1 - tan x * tan y) :=
+begin
+  cases h,
+  { rw [tan, sin_add, cos_add,
+        ← div_div_div_cancel_right (sin x * cos y + cos x * sin y)
+            (mul_ne_zero (cos_ne_zero_iff.mpr h.1) (cos_ne_zero_iff.mpr h.2)),
+        add_div, sub_div],
+    simp only [←div_mul_div, ←tan, mul_one, one_mul,
+              div_self (cos_ne_zero_iff.mpr h.right), div_self (cos_ne_zero_iff.mpr h.left)] },
+  { cases h.1 with k h1, cases h.2 with l h2,
+    have hx := tan_int_mul_pi_div_two (2*k+1),
+    have hy := tan_int_mul_pi_div_two (2*l+1),
+    have hxy := tan_int_mul_pi_div_two (2*k+1+(2*l+1)),
+    simp only [int.cast_add, int.cast_bit0, int.cast_mul, int.cast_one, ← h1, ← h2] at hx hy hxy,
+    rw [hx, hy, add_zero, zero_div, h1, h2, mul_div_assoc, mul_div_assoc,
+        ← add_mul (2*(k:ℂ)+1) (2*(l:ℂ)+1) ((π:ℂ)/2), ← mul_div_assoc, hxy] },
+end
+
+lemma tan_two_mul {z : ℂ} : tan (2 * z) = 2 * tan z / (1 - tan z ^ 2) :=
+begin
```

Comment:
Ok, I see that `tan_add`'s convoluted statement is actually useful. I still think you should make it two lemmas.

Reply:
@PatrickMassot I did this. Please let me know if it's as you intended,